src/erlcloud_sf.erl

```diff
@@ -0,0 +1,630 @@
+configure(AccessKeyID, SecretAccessKey, Host) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host)),
+    ok.
+
+-spec configure(string(), string(), string(), non_neg_integer()) -> ok.
+configure(AccessKeyID, SecretAccessKey, Host, Port) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host, Port)),
+    ok.
+
+-spec configure(string(), string(), string(), non_neg_integer(), string()) -> ok.
+configure(AccessKeyID, SecretAccessKey, Host, Port, Scheme) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host, Port, Scheme)),
+    ok.
+
+default_config() -> erlcloud_aws:default_config().
+
+%%------------------------------------------------------------------------------
+%% CreateActivity
+%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% Create Activity API:
+%% [https://docs.aws.amazon.com/step-functions/latest/apireference/API_CreateActivity.html]
+%%
+%% ===Example===
+%%
+%%------------------------------------------------------------------------------
+-spec create_activity(ActivityName  :: binary()) -> {ok, map()} | {error, any()}.
+create_activity(ActivityName) ->
+    create_activity(ActivityName, default_config()).
+
+-spec create_activity(ActivityName  :: binary(),
+                      Config        :: aws_config()) -> {ok, map()} | {error, any()}.
+create_activity(ActivityName, Config) when is_binary(ActivityName) ->
+    Req = #{<<"name">> => ActivityName},
+    step_request(Config, post, "CreateActivity", Req).
+
+%%------------------------------------------------------------------------------
+%% CreateStateMachine
+%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% Create State Machine API:
+%% [https://docs.aws.amazon.com/step-functions/latest/apireference/API_CreateStateMachine.html]
+%%
+%% ===Example===
+%%
+%%------------------------------------------------------------------------------
+-spec create_state_machine(Definition       :: map(),
+                           StateMachineName :: binary(),
+                           RoleArn          :: binary()) ->
+    {ok, map()} | {error, any()}.
+create_state_machine(Definition, StateMachineName, RoleArn) ->
+    create_state_machine(Definition, StateMachineName, RoleArn, default_config()).
+
+-spec create_state_machine(Definition       :: map(),
+                           StateMachineName :: binary(),
+                           RoleArn          :: binary(),
+                           Config           :: aws_config()) ->
+    {ok, map()} | {error, any()}.
+create_state_machine(Definition, StateMachineName, RoleArn, Config)
+        when is_binary(StateMachineName), is_binary(RoleArn), is_map(Definition) ->
+    Req = #{
+        <<"definition">>    => jsx:encode(Definition),
```

Comment:
should encode nested maps instead of letting jsx do it later?

Reply:
in fact that should be JSON-in-JSON, so we need to do it twice